### ISSUE TITLE
Support Safari extension for macOS Catalina and after

### ISSUE THREAD

**Issue by nella17** [nickvdp/RecentTabList#19]
Safari on macOS Catalina couldn't support `safariextz`.

![Screen Shot 2019-08-30 at 03 13 12 PM](https://user-images.githubusercontent.com/14134195/64000756-f1644c00-cb38-11e9-9dab-011ca826bc6e.jpg)

**Comment by nickvdp** [comment]
Yeah :| Apple discontinued support for this kind of extension. I have no plans to update it for Catalina (it would require a complete rewrite). Some of the features of the extension have been directly added to Safari in recent versions (for example, a list of recently closed tabs under the 'History' menu)

I've submitted new extension versions to Apple (with dark mode support) but they'll never get officially released on the extension site.

I'll upload a final version of the extension here & on my site, and then it's done. Hopefully someone can pick up the pieces and recreate it (or something similar) for the new extension architecture.

**Comment by jhfrontz** [comment]
Would you be willing to update it for $$$?  I just updated to Safari 13 and am drowning in all of my open windows/tabs without your extension (having to page through 45 open Safari windows, each with 1-20 open tabs, trying to remember just *where* that one tab was...)

**Comment by tylsimp** [comment]
I'd chip in $$$ to anyone willing to figure out updating it for the new API, I've been missing this extension dearly and without it have been an organizational mess getting up to 300+ tabs open with many duplicates. None of the alternatives on Chrome or Firefox seem to work nearly as well IMO, and Safari is smoother in a lot of ways to begin with.

**Comment by tomizza** [comment]
plz nick! how can we help getting RecentTabList work on macOS Catalina?

**Comment by nickvdp** [comment]
Unfortunately, I'm not going to work on this anymore (for money or otherwise 😄). Sorry!

The code is open source and the functionality isn't anything unique, so I imagine it would be pretty straightforward to duplicate in the new Safari API (though I don't know for sure).

I think this would be a pretty good project for anyone looking to learn a new skill or language. Starting a new repo / GitHub project would probably be the smart move.

If there's no one interested in doing it as a learning exercise, I would imagine you could find freelance devs to do it on upwork.com or fiverr.com (though I've never used these myself).

Hope someone steps up or you find an alternate solution soon!

**Comment by tomizza** [comment]
thanx nick for your prompt reply -- i'll pass your comment along to some educational institutions, hopefully we can find somebody skilled & interested, is that ok?

**Comment by nickvdp** [comment]
No problem :) I'd be very happy for anyone to pick up this project.
